app/views/users/show.html.erb

```diff
@@ -0,0 +1,42 @@
+<%# <h1>User Profile</h1> %>
+<div class="user-navigation">
+  <a href="<%= users_path %>">
+    <button class="back-btn">⬅</button>
+  </a>
+</div>
+<section class="user-profile-page">
+  <div class="user-profile">
+    <% if @user.photo.present? %>
+      <%= image_tag @user.photo, alt: @user.name, class: 'user-thumbnail' %>
+    <% else %>
+      <span class="user-thumbnail placeholder">No Photo</span>
+    <% end %>
+    <p><%= @user.name %></p>
+    <p>Number of Posts: <%= @user.posts.count %></p>
+  </div>
+  <div class="user-bio">
+    <h3>Bio:</h3>
+    <p><%= @user.bio %></p>
+  </div>
+
+  <div class="user-posts">
+    <% if @user.posts.any? %>
+      <ul class="user-posts-info">
+        <% @user.posts.each do |post| %>
+          <h3><%= post.title %></h3>
+          <li><%= post.text %></li>
+          <div class="counters">
+            <li>Comments:<%= post.comments_counter %></li>
+            <li>Likes:<%= post.likes_counter %></li>
+          </div>
+        <% end %>
+      </ul>
+    <% else %>
+      <p>No posts available.</p>
+    <% end %>
+
+  </div>
+  <div>
+    <%= link_to 'See All Posts', user_posts_path(@user), class: 'see-all-button' %>
```

Comment:
- [ ] When I click on **See All Posts** button, I got the error on the screenshot below hence I couldn't see all my posts and their corresponding comments. Let's kindly :wrench: the issue :+1: probably you needs the user instance variable instead :+1: 
![Screenshot from 2023-11-11 06-21-13](https://github.com/mohammadnewton/blog_app/assets/63087491/119719d2-7962-46f7-97c0-1e8521edc1de)